components/team_settings/team_settings.jsx

```diff
@@ -61,7 +61,7 @@ TeamSettings.propTypes = {
     updateSection: PropTypes.func.isRequired,
     closeModal: PropTypes.func.isRequired,
     collapseModal: PropTypes.func.isRequired,
-    team: PropTypes.object.isRequired,
+    team: PropTypes.object,
```

Comment:
Why are we removing required here?

Reply:
@sudheerDev **team** prop is not passed to TeamSettings component used  in the team_settings_modal.

Reply:
It's not being directly passed as prop into the component however it is being passed by the redux as a prop which is why you cannot remove this. You can see it in `components/team_settings/index.js` in the `mapStateToProps` function.

Reply:
@hahmadia how can we fix the typescript error (ie) team prop is required?

Reply:
@sowmiyamuthuraman Can you do
`getCurrentTeam(state) || {}` in https://github.com/mattermost/mattermost-webapp/blob/master/components/team_settings/index.js#L12 and see if that fixes your problem? Let me know! Thanks

Reply:
@hahmadia Still facing the same issue

```Property 'team' is missing in type '{ activeTab: string; activeSection: string; updateSection: (section: string) => void; closeModal: () => void; collapseModal: () => void; }' but required in type 'Pick<Pick<Pick<{ activeTab: any; activeSection: any; updateSection: any; closeModal: any; collapseModal: any; team: any; }, "updateSection" | "activeSection" | "closeModal" | "collapseModal" | "activeTab">, never> & Pick<...>, "team" | ... 2 more ... | "collapseModal">'```

Reply:
In that case I'm not sure what to say then. I assume this change is ok then to be made. @sudheerDev do you have any suggestions?

Reply:
I looked into it further and I believe you need to change it to optional only cause you aren't directly passing it as a prop and therefore typescript is gonna complain. Unless Sudheer has a solution, I think it's fine to leave it as it is.

Reply:
That is weird, but i am gonna guess it is because this is not a ts file, and it is weird we say required and we have a check for team object at the start, 

```
if (!team) {
        return null;
    }
```

Lets go ahead with the changes as this is a minor nit tbh